src/App.tsx

```diff
@@ -17,18 +62,32 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setStatus={setSelect}
+                setInput={setQuery}
+                input={query}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!areTodosLoaded ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={filteredTodos}
+                  visible={modalVisible}
+                  setTodoOnClick={setTodo}
+                  showModal={setModalVisible}
```

Comment:
The `setTodoOnClick` and `showModal` props should be named more descriptively to reflect their purpose. Consider renaming them to `onTodoSelect` and `onModalToggle` for better readability and maintainability.